### PR TITLE
feat(Masthead, PageHeader): update component logic

### LINKED ISSUE #4665: Masthead: default element for MastheadBrand
https://github.com/patternfly/patternfly/issues/3332

> My thought would be to only make the logo an `a` tag if there's an `href`. In this context an `<a>` tag without an `href` doesn't make much sense.

Referencing the [PageHeader component](https://github.com/patternfly/patternfly-react/blob/master/packages/react-core/src/components/Page/PageHeader.tsx) - since `logoComponent` is an `a` by default, it needs some sort of link attribute for that to be the appropriate element to use there. I wonder if there is a good way to detect whether the element is suitable as an `<a>` or some other element? As in if it has an `href` or sets a location on click, it's rendered as an `<a>`? If it just has a click handler, maybe it should be a `<button>` instead, or `role="button"`, since [an `<a>` without a `href` has no implicit role](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Properties). Otherwise it's rendered as a `<span>`?

I see you can pass a `logoComponent` to achieve these variations, but was curious if there is a way we could do this detection for the user and default to a `<span>` 

UPDATE: The PageHeader and it's `logoComponent` prop are deprecated in v5. But we would want to make this same change to the MastheadBrand component, so that it's only an anchor when an href is provided, otherwise it'd default to a div (and can be overriden by MastheadBrand's `component` prop)

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #4665

Adds logic to component type of Masthead and PageHeader. If an `href` is passed, the component type will be an anchor. If an `onClick` is passed, the type will be a button. Otherwise the type will be a span.

### CODE REVIEW

**Review by thatblindgeye — APPROVED** [review]

**Review by nicolethoen — COMMENTED** [review]
we likely need a codemod to either warn about this update to the default component type in these cases, or just update the component passed to 'a' to prevent breaking changes?

**Comment by kmcfaul** [comment]
@nicolethoen Opened https://github.com/patternfly/pf-codemods/issues/242
Will probably also try to get it as well.

**Review by gitdallas — APPROVED** [review]

File: packages/react-core/src/components/Masthead/MastheadBrand.tsx [hunk 1]
```diff
@@ -15,10 +15,19 @@ export interface MastheadBrandProps
...
+    if (props?.href !== undefined) {
+      Component = 'a';
+    } else if (props?.onClick !== undefined) {
+      Component = 'button';
+    } else {
+      Component = 'span';
+    }
+  }
   return (
     <Component className={css(styles.mastheadBrand, className)} tabIndex={0} {...props}>
```
Comment (mcoker) [inline packages/react-core/src/components/Masthead/MastheadBrand.tsx:32]: If this is a `<span>`, does it need a `tabindex`?

File: packages/react-core/src/components/Masthead/MastheadBrand.tsx [hunk 2]
```diff
@@ -15,10 +15,19 @@ export interface MastheadBrandProps
...
+  component,
   ...props
 }: MastheadBrandProps) => {
-  const Component = component as any;
+  let Component = component as any;
+  if (!component) {
+    if (props?.href !== undefined) {
+      Component = 'a';
+    } else if (props?.onClick !== undefined) {
+      Component = 'button';
```
Comment (mcoker) [inline packages/react-core/src/components/Masthead/MastheadBrand.tsx:26]: Looks like we don't support styling for this as a `<button>` - just want to make sure that's a valid use case before filing an issue for it. Also if it's a button, should it have `type="button"`?

**Review by mcoker — COMMENTED** [review]
lgtm! Just a couple of comments.

File: packages/react-core/src/components/Masthead/MastheadBrand.tsx (hunk 1 above)
Comment (kmcfaul) [reply packages/react-core/src/components/Masthead/MastheadBrand.tsx:32]: Probably not, I'll update it.

File: packages/react-core/src/components/Masthead/MastheadBrand.tsx (hunk 2 above)
Comment (kmcfaul) [reply packages/react-core/src/components/Masthead/MastheadBrand.tsx:26]: Removing it for now, especially if we don't currently have styling for it.

**Review by mcoker — APPROVED** [review]
L🚀TM!

**Review by tlabaj — APPROVED** [review]

**Pull request merged**